library/spdm_common_lib/libspdm_com_support.c

```diff
@@ -229,3 +233,12 @@ void libspdm_init_managed_buffer(void *m_buffer, size_t max_buffer_size)
 
     libspdm_reset_managed_buffer(m_buffer);
 }
+
+#if LIBSPDM_FIPS_MODE
+
+size_t libspdm_module_version(void)
```

Comment:
@steven-bellock , I think we can use the semantic version, align with libspdm release. 4 bytes in total.

byte3 - libspdm major version
byte2 - libspdm minor version
byte1 - libspdm patch version
byte0 - FIPS mode enable/disable

Thought?

Reply:
Byte0 should be the release status.
0x00 - unreleased and pending
0x10 - pre-release
0x20 - released

There should be a separate mechanism to determine if FIPS is enabled or not.

Reply:
Good idea to include tag release. But I dont think we need distinguish unrelease v.s. pre-release. They should be same. One bit is enough. I suggest:
BIT[0] - official release tag or no tag.
BIT[1:5] - reserved.
BIT[6] - debug build or release debug
BIT[7] - FIPS mode or not

Why we need use different mechanism for reporting?

Reply:
The library's version is completely orthogonal to whether FIPS is enabled or not.

Reply:
I would say: the Byte0 is to provide additional binary info.
What we provide is 
* official release tag or not
* release build or debug build
* FIPS mode or not

That helps people to understand what this binary is.

If you dont like FIPS info, then we should make it `reserved to 0`.

Reply:
I don't like the FIPS information in `libspdm_module_version`. So maybe we should have something more generic like
```
typedef enum {
  LIBSPDM_INFO_VERSION,
  LIBSPDM_INF_BUILD,
  LIBSPDM_INFO_FIPS,
 } libspdm_info_t;

uint32_t libspdm_get_module_info (libspdm_info_t info);
```
Then we can expand the enum as needed.

Reply:
Thanks. I am fixing for it.

Reply:
I think we can use below APIs:
```
/*
byte3 - libspdm major version
byte2 - libspdm minor version
byte1 - libspdm patch version
byte0 - libspdm alpha (office release: 0, non official release: non-zero)
*/
uint32_t libspdm_module_version(void);
```